src/app/pages/table-page/batch-actions-table.component.ts

```diff
@@ -1,51 +1,82 @@
-import {TableData} from './table-data'
 import {ModalDirective} from '../../directives/modal.directive'
+import exampleData from './exampleData.json';
+import { CommonModule } from '@angular/common';
 
 @Component({
   selector: 'app-batch-actions-table',
   template: `
     <h1>Batch Actions</h1>
-    <tds-table id="batchTable" enable-multiselect vertical-dividers="false" compact-design="false" enable-responsive="false">
+    <tds-table #table id="batchTable" multiselect vertical-dividers="false" compact-design="false" responsive="false">
       <tds-table-toolbar table-title="Batch action">
         <div slot="end">
-          <tds-button type="primary" size="sm" text="Download" (click)="download()"></tds-button>
+          <tds-button (click)="download()" type="primary" size="sm" text="Download"></tds-button>
         </div>
       </tds-table-toolbar>
-      <tds-table-header>
+      <tds-table-header multiselect (tdsSelectAll)="selectAll($event)" [allSelected]="allSelected">
         <tds-header-cell column-key="truck" column-title="Truck type"></tds-header-cell>
         <tds-header-cell column-key="driver" column-title="Driver name"></tds-header-cell>
         <tds-header-cell column-key="country" column-title="Country"></tds-header-cell>
         <tds-header-cell column-key="mileage" column-title="Mileage"></tds-header-cell>
       </tds-table-header>
-      <tds-table-body #tableBody></tds-table-body>
+      <tds-table-body>
+          <tds-table-body-row *ngFor="let row of tableData;" selected="{{ row.selected }}" (tdsSelect)="handleRowSelect(row.id, $event)">
+              <tds-body-cell cell-key="truck">
+                  {{ row.truck }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="driver">
+                  {{ row.driver }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="country">
+                  {{ row.country }}
+              </tds-body-cell>
+              <tds-body-cell cell-key="mileage">
+                  {{ row.mileage }}
+              </tds-body-cell>
+          </tds-table-body-row>
+      </tds-table-body>
     </tds-table>
     <ng-template modal-dr />
   `,
   styles: [``],
-  imports:[ModalDirective],
+  imports:[ModalDirective, CommonModule],
   standalone: true,
-  schemas: [CUSTOM_ELEMENTS_SCHEMA]
+  schemas: [CUSTOM_ELEMENTS_SCHEMA],
 })
-export class BatchActionsTableComponent implements AfterViewInit {
-  @ViewChild('tableBody', { static: false}) tableBody: ElementRef
+export class BatchActionsTableComponent {
   @ViewChild(ModalDirective, { static: true }) modalDirective!: ModalDirective
-  renderer2 = inject(Renderer2)
-  @Input() tableData: TableData[]
+  @ViewChild('table', { static: false}) tableRef: ElementRef
 
-  ngAfterViewInit(): void {
-    this.renderer2.setAttribute(this.tableBody.nativeElement, 'body-data', JSON.stringify(this.tableData))
+  tableData = exampleData;
+
+  allSelected = this.tableData.every((row) => row.selected)
```

Comment:
I think the name tableData might be a bit misleading, since it doesn't contain all table data, but only the table data that should be shown after a filter has been applied. We could rename tableData to filteredData. 

If for example the data would be edited, then exampleData would be the data that was changed first, and then the filter would be applied on that data to get the new filteredData.

And about this row - I guess its not obvious if the active selection should be on the filtered data, or also data outside of the filter. Do you expect to select all data when you select all, or just all data that you can see?

Reply:
These are things that the users will have to consider when they do their implementation.
But, does not affect our API really, just examples.